### PR TITLE
replace validation for package installation

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/176076302

### CODE REVIEW

File: internal/runners/packages/packages.go [hunk 1]
```diff
@@ -82,6 +70,16 @@ func executePackageOperation(pj *project.Project, out output.Outputer, authentic
 
 	logging.Debug("Order changed: %v", orderChanged)
 
+	// Verify that the provided package actually exists (the vcs API doesn't care)
+	_, err = model.FetchRecipe(commitID, pj.Owner(), pj.Name(), &model.HostPlatform)
+	if err != nil {
+		rerr := &inventory_operations.ResolveRecipesBadRequest{}
+		if errors.As(err, &rerr) {
```
Comment (mdrohmann) [inline internal/runners/packages/packages.go:77]: Note, that this check is rather unspecific.  The `ResolveRecipesBadRequest` error could happen for other reasons.  Its payload has nothing that specifically says that an ingredient was missing: https://activestate.slack.com/archives/C7PEK0LLU/p1608058443447900

One thought that I had to work around this, is update the suggestions function in a way such that it also checks if the name and version that are provided are actually not matching any of the suggestions.  But then we still need to know if the features are case-sensitive or not, which is probably knowledge that should live in the solver.  So, maybe we need a follow-up story for pb?

Comment (Naatan) [reply internal/runners/packages/packages.go:77]: I think for this story we'd want to just make sure that you still report the error value of FetchRecipe.

File: pkg/platform/model/recipe.go [hunk 2]
```diff
@@ -160,11 +160,11 @@ func FetchRecipe(commitID strfmt.UUID, owner, project string, hostPlatform *stri
 
 		orderBody, _ := json.Marshal(params.Order)
 		switch rrErr := err.(type) {
-		case *iop.SolveOrderDefault:
+		case *iop.ResolveRecipesDefault:
 			msg := *rrErr.Payload.Message
 			logging.Error("Could not solve order, error: %s, order: %s", msg, string(orderBody))
 			return nil, locale.WrapInputError(err, "err_solve_order", "", msg)
-		case *iop.SolveOrderBadRequest:
```
Comment (Naatan) [inline pkg/platform/model/recipe.go:167]: Nice catch!

**Review by Naatan — CHANGES_REQUESTED** [review]
Looks good! I think all that's missing is ensuring the FetchRecipe error is always reported (even when suggesting alternatives).

**Review by Naatan — DISMISSED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**